### PR TITLE
#218 - introduced TestResource

### LINKED ISSUE #218: Class to obtain test resource
Let's introduce class to obtain resource from current class loader and put this resource to provided storage. 

This functionality is required in every adapter, this class will allow to avoid code duplications and unexpected errors.

### PR DESCRIPTION
**Pull request by olenagerasimova**
Introduced `TestResource` and test for it to close #218.

### CODE REVIEW

**Comment by olenagerasimova** [comment]
@g4s8 could you please review/assign contributors as reviewers?

File: src/main/java/com/artipie/asto/test/TestResource.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+    /**
+     * Recourse as bytes.
+     * @return Bytes
+     * @checkstyle MagicNumberCheck (10 lines)
+     */
+    @SuppressWarnings("PMD.AssignmentInOperand")
+    public byte[] asBytes() {
+        try (InputStream stream = this.asInputStream();
+            ByteArrayOutputStream buffer = new ByteArrayOutputStream()) {
```
Comment (olegmoz) [inline src/main/java/com/artipie/asto/test/TestResource.java:107]: @olenagerasimova you may move `ByteArrayOutputStream` from `try`, as it does not require closing

File: src/main/java/com/artipie/asto/test/TestResource.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        storage.save(key, new Content.From(this.asBytes())).join();
+    }
+
+    /**
+     * Reads recourse and saves it to storage by given path as a key.
+     * @param storage Where to save
+     */
+    public void saveToStorage(final Storage storage) {
+        this.saveToStorage(storage, new Key.From(this.path));
+    }
```
Comment (olegmoz) [inline src/main/java/com/artipie/asto/test/TestResource.java:71]: @olenagerasimova we may simplify these method names to `saveTo`, as `Storage` presents in arguments

File: src/main/java/com/artipie/asto/test/TestResource.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+/**
+ * Test resource.
+ * @since 0.24
+ */
+public final class TestResource {
+
+    /**
+     * Relative to test resources folder resource path.
+     */
+    private final String path;
```
Comment (olegmoz) [inline src/main/java/com/artipie/asto/test/TestResource.java:46]: @olenagerasimova term `name` is usually used in Java to reference resources, see https://docs.oracle.com/javase/8/docs/api/java/lang/ClassLoader.html#getResource-java.lang.String-

File: src/test/java/com/artipie/asto/test/TestResourceTest.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    @Test
+    void readsRecourseBytes() {
+        MatcherAssert.assertThat(
+            new TestResource("test.txt").asBytes(),
+            new IsEqual<>("hello world".getBytes())
+        );
+    }
+
+    @Test
+    void readsRecourseAsStream() {
```
Comment (olegmoz) [inline src/test/java/com/artipie/asto/test/TestResourceTest.java:52]: @olenagerasimova there are typos in both methods, `Recource` instead of `Resource`

**Comment by olenagerasimova** [comment]
@olegmoz thanks, all fixed

**Comment by 0crat** [comment]
Job `gh:artipie/asto#219` is not assigned, can't get performer

**Pull request merged**